gensim/models/word2vec.py

```diff
@@ -1120,7 +1135,8 @@ def add_word(word, weights):
                     weights = fromstring(fin.read(binary_len), dtype=REAL)
                     add_word(word, weights)
             else:
-                for line_no, line in enumerate(fin):
+                for line_no in xrange(vocab_size):
```

Comment:
In the moment, the `xrange` approach was to mirror the way the binary-branch worked, where changing `vocab_size` to `limit` already had the desired effect. Considering it against `islice`, it's probably better for a mismatch to generate an error, than just silently accept fewer-than-the-declared count of vectors. 

But, thinking of this made me realize the code wasn't yet handling a `limit` > the available count (so that's now been fixed), and the `binary` path could reach a busy-hang if reaching EOF while trying to read its token char-by-char (so both it and the text-path now recognize when a read returns nothing, indicating file-end-before-declared-count-reached, and raise an EOFError).